projector-plugin/src/main/kotlin/SessionDialog.kt

```diff
@@ -0,0 +1,245 @@
+
+      rwTokenEditor.token = generatePassword()
+      roTokenEditor.token = generatePassword()
+    }
+
+    myHostsList.onChange = ::updateInvitationLinks
+    portEditor.onChange = ::updateInvitationLinks
+    rwTokenEditor.onChange = ::updateInvitationLinks
+    roTokenEditor.onChange = ::updateInvitationLinks
+
+    updateInvitationLinks()
+    setResizable(false)
+    init()
+  }
+
+  override fun createCenterPanel(): JComponent? {
+    val addressPanel = JPanel()
+    LinearPanelBuilder(addressPanel)
+      .addNextComponent(myHostsList, width = 0.5, rightGap = 35)
+      .addNextComponent(portEditor, width = 0.5)
+
+    val panel = JPanel()
+    LinearPanelBuilder(panel)
+      .addNextComponent(description, gridCount = 2, bottomGap = 5)
+      .startNextLine().addNextComponent(addressPanel, 2)
+      .startNextLine().addNextComponent(rwTokenEditor, 2)
+      .startNextLine().addNextComponent(roTokenEditor, 2)
+      .startNextLine().addNextComponent(JLabel("Invitation Links:"), gridCount = 2, topGap = 5, bottomGap = 5)
+      .startNextLine().addNextComponent(JLabel("Full Access Link:")).addNextComponent(rwInvitationLink)
+      .startNextLine().addNextComponent(roInvitationTitle).addNextComponent(roInvitationLink)
+    return panel
+  }
+
+  private fun updateInvitationLinks() {
+    rwInvitationLink.update(host, port, rwTokenEditor.token)
+    roInvitationLink.update(host, port, roTokenEditor.token)
+
+    if (rwTokenEditor.token == roTokenEditor.token) {
+      // Hide RO invitation section when tokens are equal - user will always get RW access in this case.
+      roInvitationTitle.isVisible = false
+      roInvitationLink.isVisible = false
+    } else {
+      roInvitationTitle.isVisible = true
+      roInvitationLink.isVisible = true
+    }
+  }
+
+  private class InivitationLink() : JPanel() {
+    private val copyButton = JButton(AllIcons.Actions.Copy).apply {
+      addActionListener {
+        Toolkit
+          .getDefaultToolkit()
+          .systemClipboard
+          .setContents(StringSelection(link.text), null)
+      }
+    }
+
+    private val link: JTextField = JTextField(null).apply {
+      isEditable = false
+      background = null
+      border = null
+      columns = 30
+    }
+
+    init {
+      layout = BoxLayout(this, BoxLayout.LINE_AXIS)
+      add(link)
+      add(copyButton)
+    }
+
+    fun update(host: String, port: String, token: String?) {
+      link.text = "http://${host}:${port}" + if (token == null) "" else "/?token=${token}"
+    }
+  }
+
+  private class TokenEditor(accessKind: String) : JPanel() {
+    private val requiredCheckBox: JCheckBox = JCheckBox("Require password for $accessKind access:").apply {
+      addActionListener {
+        tokenTextField.text = if (isSelected) generatePassword() else null
+        tokenTextField.isEnabled = isSelected
+        onChange?.invoke()
+      }
+    }
+    private val tokenTextField: JTextField = JTextField().apply {
+      columns = 15
```

Comment:
For me, rw and ro text fields still have different width. Could you make them identical?